### PR TITLE
[job submission] Add job submission ID + status to /api/snapshot

### LINKED ISSUE #19989: Add job submission ID + job status to /api/snapshot metadata
1. Tack on `job_submission_id` as a field in the metadata of the `JobConfig`.
2. Add a `status` field to the jobs info in the `/api/snapshot` response. This will be set to the current job status for jobs submitted via API, else null.

### PR DESCRIPTION
**Pull request by edoakes**
## Related issue number

Closes https://github.com/ray-project/ray/issues/19989

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by architkulkarni — APPROVED** [review]
Snapshot changes look good to me, thanks for updating the schema!

File: dashboard/modules/job/data_types.py [hunk 1]
```diff
@@ -9,6 +9,9 @@
 
 
 class JobStatus(str, Enum):
+    def __str__(self):
```
Comment (tchordia) [inline dashboard/modules/job/data_types.py:12]: is this necessary? doesn't this happen automatically if you extend str

Comment (edoakes) [reply dashboard/modules/job/data_types.py:12]: By default it's `JobStatus.RUNNING`, not `RUNNING`

File: dashboard/modules/snapshot/snapshot_head.py [hunk 2]
```diff
@@ -65,16 +67,25 @@ async def snapshot(self, req):
         return dashboard_utils.rest_response(
             success=True, message="hello", snapshot=snapshot)
 
+    def _get_job_status(self, metadata: Dict[str, str]) -> Optional[str]:
+        status = None
+        job_submission_id = metadata.get(JOB_ID_METADATA_KEY)
+        if job_submission_id is not None:
+            status = str(self._job_status_client.get_status(job_submission_id))
```
Comment (tchordia) [inline dashboard/modules/snapshot/snapshot_head.py:74]: is it possible that this isn't set / returns None?

File: dashboard/modules/snapshot/snapshot_head.py [hunk 3]
```diff
@@ -84,6 +95,9 @@ async def get_job_info(self):
                 "end_time": job_table_entry.end_time,
                 "config": config,
             }
+            status = self._get_job_status(metadata)
+            if status is not None:
+                entry["status"] = status
```
Comment (tchordia) [inline dashboard/modules/snapshot/snapshot_head.py:100]: i think we should always set the status, and just make it None if it doesn't exist. This way we avoid weird key errors on the other side.

File: dashboard/modules/snapshot/snapshot_head.py (hunk 2 above)
Comment (edoakes) [reply dashboard/modules/snapshot/snapshot_head.py:74]: If there's a job submission ID, this must be set

File: dashboard/modules/snapshot/snapshot_head.py (hunk 3 above)
Comment (edoakes) [reply dashboard/modules/snapshot/snapshot_head.py:100]: Ok, sure I'll figure out how to do that. That's what I had originally but jsonschema was not having it.

Comment (edoakes) [reply dashboard/modules/snapshot/snapshot_head.py]: updated

File: dashboard/modules/snapshot/snapshot_head.py (hunk 2 above)
Comment (edoakes) [reply dashboard/modules/snapshot/snapshot_head.py:74]: added comment

**Comment by edoakes** [comment]
@tchordia comments addressed PTAL

**Pull request merged**